validator/keymanager/v2/direct/import.go

```diff
@@ -33,6 +34,11 @@ func (dr *Keymanager) ImportKeystores(
 		if err != nil {
 			return err
 		}
+		for j := 0; j < i; j++ {
+			if reflect.DeepEqual(pubKeyBytes, pubKeys[j]) {
```

Comment:
What about bytes.Equal? That seems more appropriate here

Reply:
This seems inefficient as well. I would recommend to turn `pubKeys` to a map and perform one lookup per key.

Reply:
Oh yeah, I didn't realize this was a nested loop. This check is O(n^2) where a map would be O(n) only